consensus/src/chained_bft/event_processor.rs

```diff
@@ -597,16 +614,15 @@ impl<T: Payload> EventProcessor<T> {
     /// * return a VoteMsg with the LedgerInfo to be committed in case the vote gathers QC.
     ///
     /// This function assumes that it might be called from different tasks concurrently.
-    async fn execute_and_vote(&self, proposal: ProposalMsg<T>) -> failure::Result<VoteMsg> {
+    async fn execute_and_vote(&self, proposal: Block<T>) -> failure::Result<VoteMsg> {
```

Comment:
Renaming might help?

```suggestion
    async fn execute_and_vote(&self, proposal_block: Block<T>) -> failure::Result<VoteMsg> {
```